Fix #4858: WKContentWorld Sandboxing - Sandbox as much as possible

Brandon-T opened the pull request:
## Summary of Changes
**Sandboxes:**
- Reader-Mode
- LoginsHelper
- MetadataHelper
- FindInPage
- NoImageMode
- CustomSearchHandler
- WindowRenderHelper
- ResourceDownloader

**Isolates:**
- Bookmarklets

**Doesn't Sandbox (these override JS functions or define prototypes or Object.define):**
- Playlist (can be done if we use MutationObserver)
- Media Backgrounding
- BraveSearch
- Cookie Control
- Fingerprinting Protection
- Fullscreen Helper
- Youtube Adblock
- ArchiveIsCompat

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #4858

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
- Included in the ticket

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Fix JSON parsing and weird tab-view crashing.

Removed sandbox parameter

Explicitly specify WKContentWorld everywhere.
- 1st step in switching over.

Fix unit tests

Brandon-T committed:
Backup

thypon reviewed (approved):
It seems to be in line with the previous changeset.

The possibility to use `escapeArgs: false` still seems a bad idea to me.

Brandon-T committed:
Sandbox scripts and handlers

Brandon-T committed:
Fix bad rebase

Brandon-T committed:
Fix unit tests and remove a lot of usage of `escapeArgs: false`.
In the future, we will do a much better escaping and FORCE all escaping of arguments.
